components/search/search.tsx

```diff
@@ -79,6 +82,30 @@ const Search: React.FC<Props> = (props: Props): JSX.Element => {
         new SearchUserProvider(actions.autocompleteUsersInTeam),
     ]);
 
+    const isDesktop = isDesktopApp() && isServerVersionGreaterThanOrEqualTo(getDesktopVersion(), '4.7.0');
+    if (enableFindShortcut) {
+        const handleKeyDown = useCallback((e: KeyboardEvent) => {
+            if (Utils.cmdOrCtrlPressed(e) && Utils.isKeyPressed(e, Constants.KeyCodes.F)) {
+                if (isDesktop || (!isDesktop && e.shiftKey)) {
+                    if (hideSearchBar) {
+                        actions.openRHSSearch();
+                        setKeepInputFocused(true);
+                    }
+                    handleUpdateSearchTerms(`in:${currentChannelName} `);
+                    handleFocus();
+                }
+            }
+        }, [currentChannelName, hideSearchBar]);
+
+        useEffect(() => {
```

Comment:
hook should never be inside a conditional. when needed it should handle the condition inside the hook callback

Reply:
Wouldn't it make more sense to not create the hook unless it was needed at all? Seems weird to add an action that will just check a conditional every single time and then not do anything else. Or is there a performance optimization I'm missing here?

Reply:
I think the risk is with dynamic conditionals (render errors when hook order changes). Right now `enableFindShortcut` is always true, but if it ever became dynamic in the future it might cause errors.

Reply:
@calebroseland is absolutely right with this one. There is the possibility of errors when the order or number of hooks change throughout (re-)renders.